tuf/data/root.go

```diff
@@ -43,10 +43,11 @@ func NewRoot(keys map[string]PublicKey, roles map[string]*RootRole, consistent b
 
 // ToSigned partially serializes a SignedRoot for further signing
 func (r SignedRoot) ToSigned() (*Signed, error) {
-	s, err := json.MarshalCanonical(r.Signed)
+	s, err := defaultSerializer.MarshalCanonical(r.Signed)
```

Comment:
This was basically my attempt at getting more coverage by doing dependency injection, since we have to propagate JSON errors everywhere.  I'm not sure we can isolate all or most of the JSON marshalling/unmarshalling to this package alone, but if we can, we can get more coverage that way.  Also that way, the data structures can handle the marshalling with canonical JSON themselves and I think everywhere else can just call regular `json.Marshsal` (since it will call their own `MarshalJSON` function?)

Thoughts?  This may be just overkill, since we have to propagate JSON errors everywhere, and this introduces complexity just for a test. I'm happy to take it out, if we are ok with merging with slightly lower code coverage (because I'm adding lines that propagate JSON marshaling errors).